homeassistant/components/systemnexa2/config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Config flow for the SystemNexa2 integration."""
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from sn2.device import Device
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_PUSH,
+    ConfigFlow,
+    ConfigFlowResult,
+)
+from homeassistant.const import CONF_DEVICE_ID, CONF_HOST, CONF_MODEL, CONF_NAME
+from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class _DiscoveryInfo:
+    name: str
+    host: str
+    model: str
+    device_id: str
+    device_version: str
+
+
+class SN2ConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the devices."""
+
+    VERSION = 1
+
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_PUSH
```

Comment:
Setting the connection class in the config flow is legacy. Today we do that in the manifest.json file.

https://developers.home-assistant.io/docs/creating_integration_manifest/#iot-class